### PR TITLE
Migrate `color` custom properties from `v10` to `v11`

### PR DESCRIPTION
**Pull request by aaronccasanova**
### WHAT is this pull request doing?

This PR migrates `color` custom properties from `v10` to `v11`. For more information on the replacement values please refer to the [replacementMaps](https://github.com/Shopify/polaris/blob/06883086fa8ea2311f020c8301a9c83dc89a2a3c/polaris-migrator/src/migrations/v11-styles-replace-custom-property-color/v11-styles-replace-custom-property-color.ts#L5-L204) in the [v11-styles-replace-custom-property-color](https://github.com/Shopify/polaris/pull/8257) migration.

Steps:
```sh
cd polaris-react
npx @shopify/polaris-migrator@0.0.0-snapshot-release-20230314233752 v11-styles-replace-custom-property-color './src/**/*.scss'
npx prettier --write "./src/**/*.scss"
yarn lint:styles
```

### CODE REVIEW

File: polaris-react/src/styles/shared/_buttons.scss [hunk 1]
```diff
@@ -40,7 +38,7 @@
   -webkit-tap-highlight-color: transparent;
 
   &:hover {
-    background: var(--p-action-secondary-hovered);
+    background: var(--p-color-bg-hover);
```
Comment (aaronccasanova) [inline polaris-react/src/styles/shared/_buttons.scss]:
```suggestion
    background: var(--p-color-bg-subdued-hover);
```

**Review by sam-b-rose — APPROVED** [review]
Tested Storybook, Polaris Sandbox, and polaris.shopify.com. All looks good! 🎨 💯 

I did notice that the types aren't updated yet on the CodeSandbox examples, but I assume that is contingent on the v11 release.

<details>
<summary>CodeSandebox type issues with new token names</summary>
alpha-card-example
bleed-horizaontal-example
</details>

**Review by aveline — APPROVED** [review]

**Review by laurkim — APPROVED** [review]
💯

**Review by alex-page — APPROVED** [review]

**Pull request merged**